extensions/diffbot/src/knowledge-connectors/helper/chunker.ts

```diff
@@ -0,0 +1,31 @@
+import { jsonToPlainText } from "json-to-plain-text";
+import { RecursiveCharacterTextSplitter } from "langchain/text_splitter";
+import { flattie } from 'flattie';
+
+const MAX_CHUNK_SIZE = 2000;
+
+/**
+ * Splits a JSON object into text chunks, removing specified keys and adding a title to each chunk.
+ *
+ * @param jsonObj - The JSON object to split.
+ * @param chunkTitle - The title to prepend to each chunk.
+ * @param filterKeys - An array of keys to remove from the JSON object before splitting.
+ * @returns A promise that resolves to an array of string chunks.
+ */
+export const jsonSplit = async (jsonObj: any, chunkTitle: string, filterKeys: string[]): Promise<string[]> => {
+
+    // Clone the object to avoid mutating the input
+    const clonedObj = { ...jsonObj };
+    filterKeys?.forEach(key => delete clonedObj[key]);
```

Comment:
Just a note and there is no changes needed here: copying the object this way will copy every nested property, but if the nested property is an object, then the object in the new variable still references to the nested object in the old variable. Since you are only deleting properties and not modifying them you approach is fully valid. If you want to create a deep copy, then it is possible nowadays without external libraries with [structuredClone](https://developer.mozilla.org/de/docs/Web/API/Window/structuredClone)